src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -296,11 +298,17 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <RunReadyToRunCompiler CrossgenTool="@(_CrossgenTool)"
                            ImplementationAssemblies="@(_ReadyToRunImplementationAssemblies)"
+                           ShowCompilerWarnings="$(PublishReadyToRunShowWarnings)"
                            CompilationEntry="@(_ReadyToRunCompileList)"
                            ContinueOnError="ErrorAndContinue">
       <Output TaskParameter="ExitCode" PropertyName="_ReadyToRunCompilerExitCode" />
+      <Output TaskParameter="WarningsDetected" PropertyName="_ReadyToRunWarningsDetected"/>
     </RunReadyToRunCompiler>
 
+    <PropertyGroup>
+      <_ReadyToRunCompilerHasWarnings Condition="'$(_ReadyToRunWarningsDetected)' == 'true'">true</_ReadyToRunCompilerHasWarnings>
```

Comment:
Why is this a distinct property rather than just using the task output? Is it because it could get set for either invocation? If so, please comment to that effect and use distinct intermediate variables.

Reply:
added comment